shared/middleware/tables_api.rb

```diff
@@ -87,6 +103,10 @@ class TablesApi < Sinatra::Base
   delete %r{/v3/(shared|user)-tables/([^/]+)/([^/]+)/(\d+)$} do |endpoint, channel_id, table_name, id|
     dont_cache
     TableType.new(channel_id, storage_id(endpoint), table_name).delete(id.to_i)
+
+    limits = TableLimits.new(@@redis, endpoint, channel_id, table_name)
+    limits.decrement_row_count
```

Comment:
On further reflection, I agree that case won't occur.  I changed the text in the PR description and added a comment to the code explaining the ordering is important.